rpadmin/admin.go

```diff
@@ -190,20 +197,37 @@ func newAdminAPI(urls []string, auth Auth, tlsConfig *tls.Config, dialer DialCon
 	a.retryClient.Transport = transport
 	a.oneshotClient.Transport = transport
 
+	if err := a.initURLs(urls, tlsConfig, forCloud); err != nil {
+		return nil, err
+	}
+
+	return a, nil
+}
+
+const (
+	schemeHTTP  = "http"
+	schemeHTTPS = "https"
+)
+
+func (a *AdminAPI) initURLs(urls []string, tlsConfig *tls.Config, forCloud bool) error {
+	if len(a.urls) != len(urls) {
+		a.urls = make([]string, len(urls))
+	}
```

Comment:
Is the admin api supposed to be concurrency safe? If yes, we could have a race condition in the moment where we replace the URLs. Possibly we need to use a RWMutex and update the funcs that access `a.urls`

Reply:
Good callout. Added a mutex to protect the urls.